src/status_im2/contexts/chat/messages/content/audio/view.cljs

```diff
@@ -0,0 +1,204 @@
+                     audio-duration-ms)
+        time-secs  (quot
+                    (if (or @seeking-audio? (#{:playing :seeking} @player-state))
+                      (if (<= @progress 1) (* duration @progress) @progress)
+                      duration)
+                    1000)]
+    (rn/use-effect (fn [] #(destroy-player player-key)))
+    (rn/use-effect
+     (fn []
+       (when (and (some? @current-player-key)
+                  (not= @current-player-key player-key)
+                  (= @player-state :playing))
+         (play-pause-player {:player-key        player-key
+                             :player-state      player-state
+                             :progress-ref      progress
+                             :message-id        message-id
+                             :audio-duration-ms duration
+                             :seeking-audio?    seeking-audio?
+                             :user-interaction? false})))
+     [@current-player-key])
+    (if (= @player-state :error)
+      [quo/text
+       {:style               style/error-label
+        :accessibility-label :audio-error-label
+        :weight              :medium
+        :size                :paragraph-2}
+       (i18n/label :error-loading-audio)]
+      [rn/view
+       {:accessibility-label :audio-message-container
+        :style               (style/container)}
+       [rn/touchable-opacity
+        {:accessibility-label :play-pause-audio-message-button
+         :on-press            #(play-pause-player {:player-key        player-key
+                                                   :player-state      player-state
+                                                   :progress-ref      progress
+                                                   :message-id        message-id
+                                                   :audio-duration-ms duration
+                                                   :seeking-audio?    seeking-audio?
+                                                   :user-interaction? true})
+         :style               (style/play-pause-container)}
+        [quo/icon
+         (case @player-state
+           :preparing :i/loading
+           :playing   :i/pause-audio
+           :i/play-audio)
+         {:size  20
+          :color colors/white}]]
+       [:f> quo/soundtrack
+        {:style                 style/slider-container
+         :audio-current-time-ms progress
+         :player-ref            (@active-players player-key)
+         :seeking-audio?        seeking-audio?}]
+       [quo/text
+        {:style               style/timestamp
+         :accessibility-label :audio-duration-label
+         :weight              :medium
+         :size                :paragraph-2}
+        (gstring/format "%02d:%02d" (quot time-secs 60) (mod time-secs 60))]])))
+
+(defn audio-message
+  [message context]
+  (let [player-state   (reagent/atom :not-loaded)
+        progress       (reagent/atom 0)
```

Comment:
@briansztamfater, I see `progress` is passed along in this file to other functions and it's named as `progress-ref`, even though it's not a ref. Refs are non-reactive and usually start as `(atom nil)`. If that makes sense, could you rename all instances of `progress-ref` to just `progress`, for example? I was confused by the suffix `ref` when it's a reactive state :)